server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java

```diff
@@ -130,7 +133,71 @@ public ReviewPeriod update(@NotNull ReviewPeriod reviewPeriod) {
             notifyRevieweeSupervisorsByReviewPeriod(reviewPeriod.getId(), reviewPeriod.getName());
         }
 
-        return reviewPeriodRepository.update(reviewPeriod);
+        // Update the review period and get the updated period.
+        ReviewPeriod period = reviewPeriodRepository.update(reviewPeriod);
+
+        if (period.getReviewStatus() == ReviewStatus.OPEN) {
+            // If the review period has been updated and is now open, we need
+            // to create feedback requests for all involved in the review period
+            Set<ReviewAssignment> assignments =
+                reviewAssignmentRepository.findByReviewPeriodId(period.getId());
+            UUID reviewTemplateId = period.getReviewTemplateId();
+            UUID selfReviewTemplateId = period.getSelfReviewTemplateId();
+            LocalDate closeDate = period.getCloseDate().toLocalDate();
+            LocalDate selfReviewCloseDate =
+                period.getSelfReviewCloseDate().toLocalDate();
+
+            // Log template id's that were not provided to the review period.
+            // This is the reason a feedback request will not be created.
+            if (reviewTemplateId == null) {
+                LOG.warn("Review Period: " + period.getId().toString() +
+                         " does not have a review template.");
+            }
+            if (selfReviewTemplateId == null) {
+                LOG.warn("Review Period: " + period.getId().toString() +
+                         " does not have a self-review template.");
+            }
+
+            Set<UUID> selfRevieweeIds = new HashSet();
+            for (ReviewAssignment assignment : assignments) {
+                // Determine the creator of this feedback request
+                List<MemberProfile> profile =
+                    memberProfileRepository.findSupervisorsForId(
+                        assignment.getReviewerId());
+                UUID creatorId = profile.isEmpty() ?
+                    assignment.getReviewerId() : profile.get(0).getId();
+
+                // This person is being reviewed and will need a self-review
+                // request.
+                selfRevieweeIds.add(assignment.getRevieweeId());
+
+                // Create the review feedback request.
+                if (reviewTemplateId != null) {
+                    createReviewRequest(period, creatorId,
+                                        assignment.getRevieweeId(),
+                                        assignment.getReviewerId(),
+                                        reviewTemplateId, closeDate);
+                }
+            }
+
+            for(UUID memberId : selfRevieweeIds) {
+                // Determine the creator of this feedback request
+                List<MemberProfile> profile =
+                    memberProfileRepository.findSupervisorsForId(memberId);
+                UUID creatorId = profile.isEmpty() ?
+                    memberId : profile.get(0).getId();
```

Comment:
```
List<MemberProfile> profile =
    memberProfileRepository.findSupervisorsForId(memberId);
UUID creatorId = profile.isEmpty() ?
    memberId : profile.get(0).getId();
``` 
is duplicated.

Extract the logic into a helper method:

```
private UUID findCreatorId(UUID memberId) {
    List<MemberProfile> profile = memberProfileRepository.findSupervisorsForId(memberId);
    return profile.isEmpty() ? memberId : profile.get(0).getId();
}
```

Replace both duplicated sections:

```
UUID creatorId = findCreatorId(assignment.getReviewerId()); // for review requests
UUID creatorId = findCreatorId(memberId); // for self-review requests
```